### PR TITLE
Use iouring for qemu block devices

### PR DESCRIPTION
**Pull request by amshinde**
io_uring was introduced as a new kernel IO interface in kernel 5.1.
It is designed for higher performance than the older Linux AIO API.
This feature was added in qemu 5.0.

Fixes https://github.com/kata-containers/kata-containers/issues/4645

### CODE REVIEW

**Comment by fidencio** [comment]
@amshinde, apart from some small problems caught by the static checks, the PR looks good.
I'll approve it once the CI is green!  Nice work!

**Comment by liubin** [comment]
@amshinde we have many configs that can be configured by [annotations](https://github.com/kata-containers/kata-containers/blob/main/docs/how-to/how-to-set-sandbox-config-kata.md), does  `block_device_aio` need to follow this convention?

File: src/runtime/pkg/oci/utils.go [hunk 1]
```diff
@@ -683,6 +683,22 @@ func addHypervisorBlockOverrides(ocispec specs.Spec, sbConfig *vc.SandboxConfig)
...
 	}
 
+	if value, ok := ocispec.Annotations[vcAnnotations.BlockDeviceAIO]; ok {
+		supportedAIO := []string{config.AIONative, config.AIOThreads, config.AIOIOUring}
+
+		valid := false
+		for _, b := range supportedAIO {
+			if b == value {
+				sbConfig.HypervisorConfig.BlockDeviceAIO = value
+				valid = true
```
Comment (liubin) [inline src/runtime/pkg/oci/utils.go:693]: Nit: add a `break`

**Review by liubin — APPROVED** [review]

**Comment by amshinde** [comment]
@fidencio Can you take another look at this?

**Review by fidencio — APPROVED** [review]
lgtm, thanks @amshinde!

**Comment by fengwang666** [comment]
Is there any performance data comparison after upgrading to iouring?

**Pull request merged**